llama_index/vector_stores/deeplake.py

```diff
@@ -159,50 +127,26 @@ def add(self, embedding_results: List[NodeWithEmbedding]) -> List[str]:
         Returns:
             List[str]: List of ids inserted.
         """
-        data_to_injest = []
-        ids = []
+        embedding = []
+        metadata = []
+        id = []
+        text = []
 
         for result in embedding_results:
-            embedding = result.embedding
-            metadata = result.node.metadata or {}
-            metadata = {**metadata, **{"document_id": result.ref_doc_id}}
-            id = result.id
-            text = result.node.get_content(metadata_mode=MetadataMode.NONE)
-
-            data_to_injest.append(
-                {
-                    "text": text,
-                    "metadata": metadata,
-                    "ids": id,
-                    "embedding": embedding,
-                }
-            )
-            ids.append(id)
-
-        @self._deeplake.compute
-        def ingest(sample_in: List[Dict], sample_out: Any) -> None:
-            for item in sample_in:
-                sample_out.text.append(item["text"])
-                sample_out.metadata.append(item["metadata"])
-                sample_out.embedding.append(item["embedding"])
-                sample_out.ids.append(item["ids"])
-
-        batch_size = min(self.ingestion_batch_size, len(data_to_injest))
-        batched = [
-            data_to_injest[i : i + batch_size]
-            for i in range(0, len(data_to_injest), batch_size)
-        ]
-
-        ingest().eval(
-            batched,
-            self.ds,
-            num_workers=min(self.num_workers, len(batched) // self.num_workers),
+            embedding.append(result.embedding)
+            _metadata = result.node.metadata or {}
+            metadata.append({**_metadata, **{"document_id": result.ref_doc_id}})
+            id.append(result.id)
+            text.append(result.node.get_content(metadata_mode=MetadataMode.NONE))
+
+        return self.vectorstore.add(
+            embedding=embedding,
+            metadata=metadata,
+            id=id,
+            text=text,
```

Comment:
if we are inserting text and metadata now (assuming this is supported properly by deeplake), we should be using the `node_to_metadata_dict` when adding and `metadata_dict_to_node` when loading (this will help serialize and store the entire node). See some other vector stores for examples here.

Then we can change the stores_text class attribute at the top to true, so that we can persist everything in just the vector store

Reply:
@logan-markewich Thx for your comments! I found them very useful and addressed them!